ci-operator/step-registry/baremetalds/sno/setup/baremetalds-sno-setup-commands.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ baremetalds sno setup command ************"
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+# Initial check
+if [ "${CLUSTER_TYPE}" != "packet" ] ; then
+    echo >&2 "Unsupported cluster type '${CLUSTER_TYPE}'"
+    exit 1
+fi
+
+# Fetch packet server IP
+IP=$(cat "${SHARED_DIR}/server-ip")
+
+SSHOPTS=(-o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' -i "${CLUSTER_PROFILE_DIR}/.packet-kni-ssh-privatekey")
+
+# Checkout packet server
+for x in $(seq 10) ; do
+    test "$x" -eq 10 && exit 1
+    ssh "${SSHOPTS[@]}" "root@${IP}" hostname && break
+    sleep 10
+done
+
+# Copy sno source from current directory to the remote server
+tar -czf - . | ssh "${SSHOPTS[@]}" "root@${IP}" "cat > /root/sno.tar.gz"
+
+# Prepare configuration and run
+scp "${SSHOPTS[@]}" "${CLUSTER_PROFILE_DIR}/pull-secret" "root@${IP}:pull-secret"
+
+# Additional mechanism to inject sno additional variables directly
+# from a multistage step configuration.
+# Backward compatible with the previous approach based on creating the
+# sno-additional-config file from a multistage step command
+if [[ -n "${SNO_CONFIG:-}" ]]; then
```

Comment:
What's the need of yet another config variable in respect to `DEVSCRIPTS_CONFIG`?

Reply:
I guess only naming. It would be weird to have ``DEVSCRIPTS_CONFIG`` used with configuration which isn't related to devscripts.
Maybe we should rename it to ``CONFIG``/``ENV`` and use it globally? Actually I'm not sure about it

Reply:
`PACKET_CONFIG` would be best IMO